core/server/models/client.js

```diff
@@ -1,14 +1,188 @@
+
+                if (domainUpdate.domainSetsAreEqual(newDomains, currentDomains)) {
+                    return;
+                }
+
+                // Domains from the current domain array which don't exist in the new domain array should be removed
+                domainsToRemove = _.reject(currentDomains, function (currentDomain) {
+                    if (newDomains.length === 0) {
+                        return false;
+                    }
+                    return _.any(newDomains, function (newDomain) {
+                        return domainUpdate.domainsAreEqual(currentDomain, newDomain);
+                    });
+                });
+
+                // Domains from the new domain array which don't exist in the DB should be created
+                domainsToCreate = _.pluck(_.reject(newDomains, function (newDomain) {
+                    return _.any(existingDomains, function (existingDomain) {
+                        return domainUpdate.domainsAreEqual(existingDomain, newDomain);
+                    });
+                }), 'trusted_domain');
+
+                // Remove any domains which don't exist anymore
+                _.each(domainsToRemove, function (domain) {
+                    domainOps.push(domainUpdate.removeDomain(DomainModel, savedModel, domain, options));
+                });
+
+                // Loop through the new domains and either add them, attach them, or update them
+                _.each(newDomains, function (newDomain) {
+                    if (domainsToCreate.indexOf(newDomain.trusted_domain) > -1) {
+                        domainOps.push(domainUpdate.createDomain(DomainModel, savedModel, newDomain, options));
+                    }
+                });
+
+                return sequence(domainOps);
+            });
+        }
+
+        // Handle updating domains in a transaction, unless we're already in one
+        if (options.transacting) {
+            return doDomainUpdates(options);
+        } else {
+            return ghostBookshelf.transaction(function (t) {
+                options.transacting = t;
+
+                return doDomainUpdates(options);
+            }).then(function () {
+                // Don't do anything, the transaction processed ok
+            }).catch(function failure(error) {
+                errors.logError(
+                    error,
+                    'Unable to save trusted domains.',
+                    'Your client was saved, but your trusted domains were not updated.'
+                );
+                return Promise.reject(new errors.InternalServerError(
+                    'Unable to save trusted domains. Your client was saved, but your trusted domains were not updated. ' + error
+                ));
+            });
+        }
+    },
+
+    trustedDomains: function trusted_domains() {
         return this.hasMany('ClientTrustedDomain', 'client_id');
+    },
+
+    trusted_domains: function trusted_domains2() {
```

Comment:
Yes this function should be snake-case rather than camel, see the post model for an example.